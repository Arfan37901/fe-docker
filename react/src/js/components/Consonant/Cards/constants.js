
const defaultProps = {
    id: '',
    title: '',
    label: '',
    footer: '',
    ctaLink: '',
    endTime: '',
    logoAlt: '',
    logoSrc: '',
    videoURL: '',
    badgeText: '',
    startTime: '',
    cardImage: '',
    bannerIcon: '',
    description: '',
    logoBorderColor: '',
    bannerFontColor: '',
    isBookmarked: false,
    bannerDescription: '',
    disableBookmarkIco: '',
    logoBackgroundColor: '',
    bannerBackgroundColor: '',
};

export const SquareCardDefaultProps = {
    ...defaultProps,
    ctaLink: '',
};

export const FullCardDefaultProps = {
    ...defaultProps,
    ctaLink: '',
};

export const WideCardDefaultProps = {
    ...defaultProps,
    footer: [],
    isBookmarked: false,
    disableBookmarkIco: false,
};
