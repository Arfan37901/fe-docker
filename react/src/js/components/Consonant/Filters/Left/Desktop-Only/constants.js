export const selectedItemsDefaultProps = {
    numItemsSelected: 0,
};

export const clearButtonDefaultProps = {
    clearAllFiltersText: '',
};

export const titleDefaultProps = {
    panelHeader: '',
};
