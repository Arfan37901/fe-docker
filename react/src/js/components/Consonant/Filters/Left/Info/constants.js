
/**
 * Minimal viewport width to fit desktops/laptops
 * @type {Number}
 */
export const DESKTOP_MIN_WIDTH = 1200;

export const defaultProps = {
    cardsQty: 0,
    filtersQty: 0,
    selectedFiltersQty: 0,
    windowWidth: window.innerWidth,
};
