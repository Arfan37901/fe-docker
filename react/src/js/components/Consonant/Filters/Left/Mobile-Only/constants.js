export const groupFooterDefaultProps = {
    numItemsSelected: 0,
    clearFilterText: '',
    mobileGroupTotalResultsText: '',
};

export const panelFooterDefaultProps = {
    resQty: 0,
    doneText: '',
    applyText: '',
    showTotalResults: false,
    clearAllFiltersText: '',
    showTotalResultsText: '',
    someFiltersAreSelected: false,
};

export const titleDefaultProps = {
    leftPanelMobileHeader: '',
};

export const infoDefaultProps = {
    selectedFiltersQty: 0,
    mobileFilterBtnLabel: '',
};
