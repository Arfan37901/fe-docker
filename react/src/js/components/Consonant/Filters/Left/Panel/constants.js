
/**
 * Minimal viewport width to fit desktops/laptops
 * @type {Number}
 */
export const DESKTOP_MIN_WIDTH = 1200;

export const defaultProps = {
    resQty: 0,
    filters: [],
    selectedFiltersQty: 0,
    showMobileFilters: false,
    windowWidth: window.innerWidth,
};
