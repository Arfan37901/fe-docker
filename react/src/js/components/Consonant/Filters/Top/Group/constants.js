export const defaultProps = {
    numItemsSelected: 0,
    clearFilterText: '',
};

/**
 * Minimum quantity of the top filter options to apply blur on options' wrapper
 * @type {Number}
 */
export const clipWrapperItemsCount = 9;
