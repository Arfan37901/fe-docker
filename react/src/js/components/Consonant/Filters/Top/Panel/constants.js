/**
 * Top search bar identifier
 * @type {String}
 */
export const searchId = 'top-search';


export const defaultProps = {
    resQty: 0,
    filters: [],
    showLimitedFiltersQty: false,
};
