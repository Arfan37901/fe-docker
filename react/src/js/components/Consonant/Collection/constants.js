import { DEFAULT_SHOW_ITEMS_PER_PAGE } from '../Helpers/constants';

// eslint-disable-next-line import/prefer-default-export
export const defaultProps = {
    pages: 1,
    cards: [],
    resultsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
};
