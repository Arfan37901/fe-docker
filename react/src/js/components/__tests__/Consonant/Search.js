import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Search from '../../Consonant/Search';

import { DEFAULT_PROPS } from '../constants/search';

import makeSetup from '../utils/settings';

Enzyme.configure({ adapter: new Adapter() });

const getSetup = makeSetup(Search, DEFAULT_PROPS);

describe('Consonant/Search', () => {
    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const { tree } = getSetup();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onBlur', () => {
            const { props: { onBlur }, wrapper } = getSetup();

            const inputElement = wrapper.find('input');

            inputElement.simulate('focus');
            inputElement.simulate('blur');

            expect(onBlur).toBeCalled();
        });
        test('should call onSearch', () => {
            const { props: { onSearch }, wrapper } = getSetup();

            wrapper.find('input').simulate('change', { target: { value: 'Search string' } });

            expect(onSearch).toBeCalled();
            expect(onSearch).toBeCalledWith('Search string');
            onSearch.mockClear();
        });
        test('should clear search value', () => {
            const { props: { onSearch }, wrapper } = getSetup();

            wrapper.find('.consonant-search--input-clear').simulate('click');

            expect(onSearch).toBeCalled();
            expect(onSearch).toBeCalledWith('');
            onSearch.mockClear();
        });
    });
});
