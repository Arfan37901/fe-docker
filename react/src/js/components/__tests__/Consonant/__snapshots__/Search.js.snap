// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Search Check snapshots should renders correctly 1`] = `
<div
  className="consonant-search"
>
  <label>
    <span
      className="consonant-search--input-title"
    >
      Search
    </span>
    <span
      className="consonant-search--input-wrapper"
    >
      <input
        onBlur={[MockFunction]}
        onChange={[Function]}
        placeholder="search text"
        required={true}
        type="search"
        value=""
      />
      <button
        className="consonant-search--input-clear"
        onClick={[Function]}
        title="Click to clear search query"
        type="button"
      >
        clear
      </button>
    </span>
  </label>
</div>
`;
